File: test/org.eclipse.vjet.test.core.jstojava/vjLibTestFiles/vjoPro/samples/basic/deploymentoptions/jstype/Sample.js
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
vjo.ctype('vjoPro.samples.basic.deploymentoptions.jstype.Sample') //< public
.protos({

/**
* @return void
* @access public
* @param {String} pHello
*/
//> public constructs(String pHello)
constructs:function(pHello){

alert("constructed with " + pHello);
/**
* @return void
* @access protected
*/
this.innerFuncOne = function(){
alert("innerFunctionOne");

};

/**
* @return void
* @access protected
*/
this.innerFuncTwo = function(){
alert("innerFunctiontwo");

};

x:{function(){alert("test");}},
innerFuncOne : function(){

},
/**
* @return void
* @access public
* @JsEventHandler
*/
//> public void hello()
hello : function(){
this.printArgs(arguments);
}

}).inits(
function(){
document.write("calling init from built file");
}
)
.endType();

File: core/test/org.eclipse.vjet.test.core.javatojs/src/org/eclipse/vjet/dsf/javatojs/tests/data/Sample.java
/*******************************************************************************
 * Copyright (c) 2005-2012 eBay Inc.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 *******************************************************************************/
package org.eclipse.vjet.dsf.javatojs.tests.data;

import java.util.Date;

import org.eclipse.vjet.dsf.dap.rt.JsBase;
import org.eclipse.vjet.dsf.html.dom.HtmlTypeEnum;
import org.eclipse.vjet.dsf.jsnative.HtmlElement;

public class Sample extends JsBase {
	
	private final String m_message;
	
	public Sample(String message) {
		m_message = message;
	}
	
	public void update(HtmlElement element){
		HtmlElement span = document().createElement(HtmlTypeEnum.SPAN.getName());
		span.setInnerHTML(getMessage());
		element.appendChild(span);
	}
	
	private String getMessage() {
		Date now = new Date();
		return m_message+" and now is "+now.toString();
	}
}
